### PR TITLE
Support multiple field/value pairs in cslib_object set

### PR DESCRIPTION
**Pull request by CareyJWilliams**
Hi @StephenXHart,

Please could you let me know what you think of this?

### CODE REVIEW

**Comment by StephenXHart** [comment]
It's the logical extension and is backward compatible and I like it.
I didn't understand lines 155-156:
*temp p_field param[p_param]
*temp p_value param[p_param+1]
Can you do that in CS? I'd have thought you need something like {param&"_${p_param}"}

**Comment by CareyJWilliams** [comment]
> It's the logical extension and is backward compatible and I like it.
> I didn't understand lines 155-156:
> *temp p_field param[p_param]
> *temp p_value param[p_param+1]
> Can you do that in CS? I'd have thought you need something like {param&"_${p_param}"}

There is limited "array notation", yeah. It doesn't always work where you want it to, but for simple statements it's a little easier to read than the dereferenced equivalents.

It's basically short hand for '_{THING}', for example:

```
*create myvar_1 "Hello"
*create myvar_2 "World"
*temp index 2
${myvar[1]} == "Hello"
${myvar[index]} == "World"
```

It also works with text:

```
*create player_fname "Obi"
*create player_mname "Wan"
*create player_lname "Kenobi"
*temp fav_name "lname"
Help ${player["fname"]} ${player["mname"]} ${player["lname"]}, you're my only hope.

Also known as: General ${player[fav_name]}
```

File: scenes/test_object.txt [hunk 1]
```diff
@@ -57,6 +57,24 @@
 *gosub test_assert (dragon_name = dragon_2_name) "=> Expected - Locke[n/]=> Received - ${dragon_name}[n/]"
 *gosub test_finish
 
+*gosub test_start "set multiple parameters in one call"
+*gosub_scene cslib_object set "dragon" 2 "name" "Mr. Multiple" "hit_points" 999
+*gosub_scene cslib_object get "dragon" 2
+*gosub test_assert_equal "Mr. Multiple" dragon_name "=> Expected - Mr. Multiple[n/]=> Received - ${dragon_name}[n/]"
+*gosub test_assert_equal dragon_hit_points 999 "=> Expected - 900[n/]=> Received - ${dragon_hit_points}[n/]"
```
Comment (StephenXHart) [inline scenes/test_object.txt]: Message says 900 - should be 999

**Review by StephenXHart — APPROVED** [review]
Fix assert message (900 should be 999) otherwise LGTM

File: scenes/test_object.txt (hunk 1 above)
Comment (CareyJWilliams) [inline scenes/test_object.txt]:
```suggestion
*gosub test_assert_equal dragon_hit_points 999 "=> Expected - 999[n/]=> Received - ${dragon_hit_points}[n/]"
```
Woops! Good catch :)

**Pull request merged**